core/src/main/scala/org/apache/spark/sql/execution/row/RowFormatRelation.scala

```diff
@@ -160,6 +160,19 @@ class RowFormatRelation(
       connection.close()
     }
   }
+  override protected def constructSQL(indexName: String,
+                           baseTable: String,
+                           indexColumns: Seq[String],
+                           options: Map[String, String]): String = {
+
+    val parameters = new CaseInsensitiveMutableHashMap(options)
+    val columns = indexColumns.reduceLeft[String](_ + "," + _)
+    val indexType = parameters.get(ExternalStoreUtils.INDEX_TYPE) match {
```

Comment:
Hm, reduceLeft vs mkString. I think latter will be more efficient in general because of the use of a single StringBuilder so I would always prefer mkString for string concat with separator (though efficiency is not important here).